<launch>
  <!-- TO CONVERT -->
  <!-- Parameter definitions -->
  <param name="prompt" value="floor.chair.waterbottle" />
  <param name="target" value="chair" />

  <!-- Arguments to enable or disable nodes -->
  <arg name="run_img_display" default="true" />
  <arg name="run_img_save" default="false" />
  <arg name="run_mask_display" default="true" />

  <!-- Always run Nodes-->
  <node name="kinect_pub_node" type="kinect_pub_node" pkg="kinect_pub" />
  <node name="img_processor_node" pkg="image_processing" type="img_processor_node.py" output="screen" />
  <!-- <node name="obstacle_avoidance_node" pkg="obstacle_avoidance" type="obstacle_avoidance.py" output="screen"/> -->

  <!-- Node definitions with conditionals -->
  <node if="$(arg run_img_display)" name="img_display_node" pkg="image_processing" type="img_display_node.py" output="screen" />
  <node if="$(arg run_img_save)" name="img_save_node" pkg="image_processing" type="img_save_node.py" output="screen" />
  <node if="$(arg run_mask_display)" name="mask_display_node" pkg="image_processing" type="mask_display_node.py" output="screen" />

</launch>
